### PR TITLE
Drop nested Promises

### PR DESCRIPTION
**Pull request by fregante**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,13 +3,13 @@ module.exports = (promise, onFinally) => {
...
 
 	return promise.then(
-		val => new Promise(resolve => {
-			resolve(onFinally());
-		}).then(() => val),
-		err => new Promise(resolve => {
-			resolve(onFinally());
-		}).then(() => {
+		val => {
+			onFinally();
```
Comment (julien-f) [inline index.js]: Nope, it's necessary to wait for the `onFinally()` completion.

**Comment by fregante** [comment]
Too bad async/await is Node 7.6+. This works:

```js
'use strict';
module.exports = async (promise, onFinally) => {
	try {
		return await promise;
	} finally {
		if (onFinally) {
			await onFinally();
		}
	}
};
```

**Comment by sindresorhus** [comment]
I think this would add another tick to the execution as the executor in the Promise constructor is called immediately, while `Promise.resolve()` is not.

**Comment by julien-f** [comment]
@sindresorhus I don't think that's true, both `new Promise()` and `Promise.resolve()` are called immediately.

And in both case, the callback in `then()` is called in the next microtask.

**Comment by fregante** [comment]
If this is the right way to test this, then this PR doesn't change the output:

```js
const pFinally = require('.');

Promise.resolve('done1').then(console.log)
pFinally(Promise.resolve('done2'), () => console.log('finally')).then(console.log)


let i = 0;
(function loop() {
	console.log(i++); // Display tick number
	if (i < 4) {
		setImmediate(loop);
	}
})();

```
```
0
done1
finally
done2
1
2
3
```

**Comment by sindresorhus** [comment]
Alright. I can't remember where I read that, so I must remember it incorrectly.

**Pull request merged**